### PR TITLE
[public-api] Only run breaking detector when public api defs change

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: .github/workflows/permissions.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 name: permissions
-on: pull_request
+on:
+  pull_request:
```
Comment (easyCZ) [inline .github/workflows/permissions.yml:3]: Drive-by syntax fix

**Review by vulkoingim — APPROVED** [review]
